### PR TITLE
tickets/DM 38375

### PR DESCRIPTION
**Pull request by couger01**
- Remove branch argument
- Use named parameters in map

### CODE REVIEW

**Review by r-owen — APPROVED** [review]
Thank you for this useful change. I don't know enough Groovy to do a thorough review, so my apologies if I missed anything.

File: vars/DevelopPipeline.groovy [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-def call(name, idl_name, module_name, pre_commit_flags="", required_idl=[], build_all_idl=false, extra_packages=[]) {
+def call(Map pipeline_args = [:]) {
     // create a developer build pipeline
+    defaultArgs = [pre_commit_flags: "", required_idl: [], build_all_idl: false, extra_packages: []]
+    pipeline_args = defaultArgs << pipeline_args
     Csc csc = new Csc()
-    idl_string = "${idl_name} "
-    if (!required_idl.isEmpty()) {
-        required_idl.each { idl ->
+    idl_string = "${pipeline_args.idl_name} "
```
Comment (wvreeven) [inline vars/DevelopPipeline.groovy:11]: What happens if the `idl_name` arg is missing from the map? I think the job should be marked as failed with an error message indicating that that parameter is mandatory. Same for `module_name`.

**Review by wvreeven — CHANGES_REQUESTED** [review]
Please make sure that the job fails if the two mandatory arguments are missing from the map.

File: vars/DevelopPipeline.groovy [hunk 2]
```diff
@@ -1,11 +1,16 @@
 import org.lsst.ts.jenkins.components.Csc
 
-def call(name, idl_name, module_name, pre_commit_flags="", required_idl=[], build_all_idl=false, extra_packages=[]) {
+def call(Map pipeline_args = [:]) {
     // create a developer build pipeline
+    defaultArgs = [pre_commit_flags: "", required_idl: [], build_all_idl: false, extra_packages: []]
+    pipeline_args = defaultArgs << pipeline_args
+    if((!pipeline_args["name"]) || (!pipeline_args["idl_name"]) || (!pipeline_args["module_name"])) {
+        error "Need to define name, idl_name or module_name."
```
Comment (wvreeven) [inline vars/DevelopPipeline.groovy]: `or` -> `and`

**Review by wvreeven — APPROVED** [review]
Looks good to me.

**Pull request merged**